packages/sdk/react-client/src/testing/ProfileInitializer.tsx

```diff
@@ -0,0 +1,34 @@
+//
+// Copyright 2020 DXOS.org
+//
+
+import React, { ReactNode, useEffect } from 'react';
+
+import { useClient } from '../client';
+import { useIdentity } from '../halo';
+
+let didInit = false;
+
+/**
+ * Automatically creates a random DXOS profile.
+ */
+// TODO(burdon): Replace with useTestProfile?
+export const ProfileInitializer = ({ children }: { children: ReactNode }) => {
```

Comment:
I like the pattern of providing the hook (most of below) then wrapping with an HOC.
(see withReactor and useReactor).